src/ua_types_encoding_binary.c

```diff
@@ -778,8 +786,15 @@ ExtensionObject_decodeBinary(UA_ExtensionObject *dst, const UA_DataType *_) {
         /* helping clang analyzer, typeId is numeric */
         UA_assert(typeId.identifier.byteString.data == NULL);
         UA_assert(typeId.identifier.string.data == NULL);
-        typeId.identifier.numeric -= UA_ENCODINGOFFSET_BINARY;
-        findDataType(&typeId, &type);
+        size_t i;
+        for (i = 0; i<UA_TYPES_COUNT; i++) {
```

Comment:
This loop checks against `binaryEncodingId` not (as findDataType) against typeId.identifier.numeric.

But for the loop above (line 711) you are right.

I'll extract it to a separate method and avoid double-loops. But first I wanted to see if this solution would be OK in general.